src/alexandria3k/__main__.py

```diff
@@ -99,19 +100,102 @@ def get_data_source_instance(args):
     return class_(data_location, sample, args.attach_databases)
 
 
+def download(args):
+    """Download data using the specified data source."""
+    args.validate_args(args)
+    data_source_instance = get_data_source_instance(args)
+    if not hasattr(data_source_instance, "download"):
+        raise Alexandria3kError(
+            f"The data source {args.data_name} does not support downloading"
+        )
+    data_source_instance.download(
+        args.database, args.sql_query, args.output_path, *args.extra_args
+    )
+    perf.log(f"Data downloaded and saved to {args.output_path}")
+
+
+def validate_args(args):
+    """Validate that both database and sql_query are either both provided or both omitted."""
+    if (args.database and not args.sql_query) or (
+        args.sql_query and not args.database
+    ):
+        raise argparse.ArgumentTypeError(
+            "Both --database and --sql-query must be provided together or not at all."
+        )
+    return args
+
+
+def add_subcommand_download(subparsers):
+    """Add the arguments of the download subcommand."""
+    parser = subparsers.add_parser(
+        "download", help="Download data using the specified data source."
+    )
+    parser.set_defaults(func=download)
+    parser.add_argument(
+        "database", help="File path of the database to use", nargs="?"
+    )
+    parser.add_argument(
+        "data_name",
+        choices=facility_names("data_sources"),
+        help="Name of the data source to use",
+    )
+    parser.add_argument(
+        "-s",
+        "--sql-query",
+        type=str,
+        help="SQL query to retrieve the data for downloading",
+    )
+    parser.add_argument(
```

Comment:
This should be a required positional argument to be consistent with the `populate` method.